### PR TITLE
chore: update postcss-selector-parser

### PR DESCRIPTION
**Pull request by anikethsaha**
updating postcss-selector pasert to latest stable version (6.0.2)

### CODE REVIEW

**Comment by anikethsaha** [comment]
I will submit a fresh force push for this. Seems like something is not good with my local setup. Need to do a fresh setup for this.

**Comment by anikethsaha** [comment]
tests fixture updated because there were two breaking xhanges from `postcss-selector-parser` and one of them were at `v4.0.0` of escaping attributes selectors and spaces with `\`


[`v3 playground`](https://codesandbox.io/s/cssnano-minify-selector-selector-parser-v3-32lb1)

[`v6 playground`](https://codesandbox.io/s/cssnano-minify-selector-me7gz)

**Comment by anikethsaha** [comment]
cc @evilebottnawi

**Comment by evilebottnawi** [comment]
Please fix conflicts

**Comment by anikethsaha** [comment]
cc @evilebottnawi

**Comment by anikethsaha** [comment]
cc @evilebottnawi

**Comment by anikethsaha** [comment]
@evilebottnawi can you take a look

File: packages/cssnano/src/__tests__/postcss-minify-selectors.js [hunk 1]
```diff
@@ -53,18 +53,21 @@ test(
   'should preserve the universal selector in attribute selectors',
   processCss(
     'h1[class=" *.js "] + *.js{color:#00f}',
-    'h1[class=" *.js "]+.js{color:#00f}'
+    'h1[class=\\*\\.js]+.js{color:#00f}'
```
Comment (evilebottnawi) [inline packages/cssnano/src/__tests__/postcss-minify-selectors.js:56]: Bug :disappointed:

File: packages/cssnano/src/__tests__/postcss-minify-selectors.js [hunk 2]
```diff
@@ -53,18 +53,21 @@ test(
...
-  processCss('[filename="*.js"]{color:#00f}', '[filename="*.js"]{color:#00f}')
+  processCss('[filename="*.js"]{color:#00f}', '[filename=\\*\\.js]{color:#00f}')
 );
 
 test(
   'should preserve the universal selector in file globs',
-  processCss('[glob="/**/*.js"]{color:#00f}', '[glob="/**/*.js"]{color:#00f}')
+  processCss(
+    '[glob="/**/*.js"]{color:#00f}',
+    '[glob=\\/\\*\\*\\/\\*\\.js]{color:#00f}'
```
Comment (evilebottnawi) [inline packages/cssnano/src/__tests__/postcss-minify-selectors.js:69]: Also bug :disappointed:

File: packages/cssnano/src/__tests__/postcss-minify-selectors.js [hunk 3]
```diff
@@ -79,7 +82,7 @@ test(
   'should handle deep combinators',
   processCss(
     'body /deep/ .theme-element{color:#00f}',
-    'body /deep/ .theme-element{color:#00f}'
+    'body/deep/.theme-element{color:#00f}'
```
Comment (evilebottnawi) [inline packages/cssnano/src/__tests__/postcss-minify-selectors.js:85]: Bug

File: packages/cssnano/src/__tests__/postcss-minify-selectors.js [hunk 4]
```diff
@@ -135,7 +138,7 @@ test(
   'should normalise attribute selectors (2)',
   processCss(
     'a[class^="options["]:after{color:#00f}',
-    'a[class^="options["]:after{color:#00f}'
+    'a[class^=options\\[]:after{color:#00f}'
```
Comment (evilebottnawi) [inline packages/cssnano/src/__tests__/postcss-minify-selectors.js:141]: Looks like bug too

File: packages/cssnano/src/__tests__/postcss-minify-selectors.js [hunk 5]
```diff
@@ -159,7 +162,7 @@ test(
   'should not be responsible for normalising comments',
   processCss(
     'h1 /*!test comment*/, h2{color:#00f}',
-    'h1 /*!test comment*/,h2{color:#00f}'
+    'h1,h2,/*!test comment*/{color:#00f}'
```
Comment (evilebottnawi) [inline packages/cssnano/src/__tests__/postcss-minify-selectors.js:165]: Bug

**Comment by evilebottnawi** [comment]
Let's fix it on `postcss-selector-parser` side

**Comment by anikethsaha** [comment]
Ok I will mark is as upstream and hold.

**Comment by DawidJanczak** [comment]
Hello! I just wanted to check if there's any progress on this as it's now a bit of a higher priority because of https://github.com/advisories/GHSA-ff7x-qrg7-qggm. Old `selector-parser` uses `dot-prop^4.1.1` and so blocks the update.

**Comment by dotnetCarpenter** [comment]
> Hello! I just wanted to check if there's any progress on this as it's now a bit of a higher priority because of https://github.com/advisories/GHSA-ff7x-qrg7-qggm. Old `selector-parser` uses `dot-prop^4.1.1` and so blocks the update.

Reference: https://github.com/postcss/postcss-selector-parser/issues/230

**Comment by dotnetCarpenter** [comment]
@evilebottnawi and @anikethsaha I looked through the bug reports over at [postcss-selector-parser](https://github.com/postcss/postcss-selector-parser/issues?q=is%3Aopen+is%3Aissue+label%3Abug). Is that the bugs you are referring to here?

We use _cssnano_ and have updated _dot-prop_ manually to 5.3.0, which seems to work fine. Does all of the bugs need to be fixed upstream before changing _dot-prop_ in package.json? I realise it is not ideal to specify a third-party dependency in this project but it seems that the bugs will take a while to be fixed.

**Comment by anikethsaha** [comment]
Not sure, we can create a issue if required.

**Comment by ludofischer** [comment]
All packages are on the latest postcss-selector-parser version as of https://github.com/cssnano/cssnano/commit/9bce2b6ccfa49c50be05efb0c80c1e3938b650b3

**Pull request closed without merging**